apis/r/DESCRIPTION

```diff
@@ -33,7 +33,7 @@ Imports:
     Matrix,
     stats,
     bit64,
-    tiledb (>= 0.18.0),
+    tiledb (== 0.18),
```

Comment:
Can't do in `DESCRIPTION`.  

```sh
[... earlier lines from package check omitted ... ]
✔  checking package namespace information ...
E  checking package dependencies (1.7s)
   Package required and available but unsuitable version: ‘tiledb’
   
   See section ‘The DESCRIPTION file’ in the ‘Writing R Extensions’
   manual.
   
   See
     ‘/tmp/file2fc4054d83de9/tiledbsoma.Rcheck/00check.log’
   for details.
   
   
$ grep " tiledb " DESCRIPTION 
    tiledb (== 0.18.0),
$ 
```


Could approximate in `configure` and error if we wanted to.

Reply:
@eddelbuettel @aaronwolen @mojaveazure if anyone has priors on how to approximate in `configure` I'd be grateful. (As a non-R-expert, I'm googling now ...)

Reply:
Here is what I just did in tiledb-r to see 'am I at 2.15.0 or not' (which has deprecation implications, so i 'silence' compilation a little).  Recall the joker we hold that the script `configure` can be anything that runs when in mode 0755 so could be `/bin/sh` (but not `/bin/bash`, do.not.get.me.started....).  In this case it is `autoconf` but again that is _not_ a requirement:

```sh
## If we have pkg-config, use it to test minimal version
if test x"${have_pkg_config}" != x"no"; then
    AC_MSG_CHECKING([for pkg-config checking TileDB version])
    if pkg-config --atleast-version=2.0.0 tiledb; then
        AC_MSG_RESULT([>= 2.0.0])
    else
        AC_MSG_RESULT([insufficient])
        AC_MSG_ERROR([TileDB Embedded 2.0.0 or later is required.])
    fi
    AC_MSG_CHECKING([for pkg-config checking TileDB deprecation silencer])
    if pkg-config --atleast-version=2.15.0 tiledb; then
        ## For TileDB 2.15 or newer are fine
        AC_MSG_RESULT([not required])
        AC_SUBST([TILEDB_SILENT_BUILD], "")
    else
        ## For TileDB older than 2.15 we use this to silence a deprecation warning
        AC_MSG_RESULT([enabled])
        AC_SUBST([TILEDB_SILENT_BUILD], "-DTILEDB_SILENT_BUILD")
    fi
fi
```

Reply:
And I should have checked first we already _do_ have `configure` as a shell script in `apis/r`.

Reply:
And I was absent-minded.  `pkg-config` is to check TileDB Embedded.  You want to check the _R_ package.  For that we would use something akin to

```sh
$ Rscript -e 'packageVersion("tiledb") == "0.18.0"'          # I am at 0.18.0.1 even though CRAN is at 0.18.0
[1] FALSE
$ Rscript -e 'packageVersion("tiledb") >= "0.18.0" && packageVersion("tiledb") < "0.19.0"'
[1] TRUE
$ 
```

Reply:
Or even

```sh
$ Rscript -e 'tiledb::tiledb_version(TRUE) == "2.15.0"'
[1] TRUE
$ 
```